src/endpoint/sts/ops/sts_post_assume_role.js

```diff
@@ -49,11 +52,52 @@ async function assume_role(req) {
 }
 
 // create and return the signed token
+/**
+ * @param {Object} auth_options
+ * @param {Number} expiry in seconds
+ * @returns {String}
+*/
 function generate_session_token(auth_options, expiry) {
     dbg.log1('sts_post_assume_role.make_session_token: ', auth_options, expiry);
     return jwt_utils.make_auth_token(auth_options, { expiresIn: expiry });
 }
 
+// TODO: Generalize and move to a utils file in the future
+/**
+ * @param {String|undefined} duration_input
+ * @returns {Number} duration in milliseconds
```

Comment:
Need to be consistent because you made this function return duration in seconds and still return milliseconds from config.STS_DEFAULT_SESSION_TOKEN_EXPIRY_MS in the default case.

I would naturally prefer durations to be in millis internally, and only convert to duration_sec on demand - which would be when calling  generate_session_token() above.